File: sample/timer_event.cpp
#include <event/dispatcher.hpp>
#include <event/timer.hpp>
#include <iostream>

int main(int argc, char* argv[])
{
	if (argc != 3)
	{
		std::cerr << "Usage: timer_event <period> <tick_cnt>" << std::endl;
		return -1;
	}

	float period = atof(argv[1]);
	int tick_cnt = atoi(argv[2]);

	ranger::event::dispatcher disp;
	ranger::event::timer tmr(disp, [period, &tick_cnt] (ranger::event::timer& tmr)
			{
				if (tick_cnt-- > 0)
				{
					std::cout << "tick: " << tick_cnt << std::endl;
					tmr.active(std::chrono::duration<float>(period));
				}
			});
	tmr.active(std::chrono::duration<float>(period));

	return disp.run();
}

File: sample/timer_event.lua
require "lua_ranger_event"
event = lua_ranger_event

tick_cnt = 10

disp = event.dispatcher()
tmr = event.timer(disp)
tmr:set_event_handler(function(self)
	if tick_cnt > 0 then
		tick_cnt = tick_cnt - 1
		print("tick:", tick_cnt)
		tmr:active(1)
	end
end)

tmr:active(1)
disp:run()
